oh-my-zsh.sh

```diff
@@ -59,9 +59,54 @@ if [ -z "$ZSH_COMPDUMP" ]; then
   ZSH_COMPDUMP="${ZDOTDIR:-${HOME}}/.zcompdump-${SHORT_HOST}-${ZSH_VERSION}"
 fi
 
-# Load and run compinit
-autoload -U compinit
-compinit -i -d "${ZSH_COMPDUMP}"
+# If one or more completion-dependent directories have insecure ownership or
+# permissions, the low-level compaudit() function fails with output resembling:
+#
+#     There are insecure directories:
+#     /usr/share/zsh/site-functions
+#     /usr/share/zsh/5.0.6/functions
+#     /usr/share/zsh
+#     /usr/share/zsh/5.0.6
+#
+# Silently detect and, in a few well-known cases, offer to fix this issue.
+if ! compaudit &>/dev/null; then
+  # Print this list of insecure directories.
+  compaudit
+  print
+
+  # If this is Cygwin *and* both standard input and output are connected to an
+  # interactive terminal, offer to fix this. Cygwin has a long-standing bug in
```

Comment:
There is this one in the mailing list: [Zsh completion problem](https://www.cygwin.com/ml/cygwin/2011-03/msg00748.html) but it looks like [it was solved](https://www.cygwin.com/ml/cygwin-announce/2011-01/msg00006.html).

Also, [in this blog post](http://blog.en.edaro.net/2014/01/fix-broken-autocompletion-with-zsh-in.html) the author says this one-liner didn't fix it for him because, basically, the Windows ACLs weren't properly set up for the user that run cygwin.

So it looks to me that this PR will need a little more testing. @leycec can you direct some babun users this way? Let's see if they can confirm this solves compinit problems in a fresh installation of babun/cygwin.